src/functionalTest/java/uk/gov/hmcts/reform/sendletter/config/MockRedisConfiguration.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.sendletter.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.GenericContainer;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@Configuration
+public class MockRedisConfiguration {
+
+    private static final Logger log = LoggerFactory.getLogger(MockRedisConfiguration.class);
+
+    static class MockRedisServer {
+
+        @Value("${redis.port}")
+        private int port;
+
+        private GenericContainer container;
+
+        @PostConstruct
+        public void init() {
+            container = new FixedHostPortGenericContainer("redis:4")
+                .withFixedExposedPort(port, port);
+            container.start();
```

Comment:
would be nice to not do all that when `redis.enabled` is set to `false`

Reply:
it is not. below the bean checks flag and if it is on - returns instance of redis container, null otherwise